mathics/builtin/patterns.py

```diff
@@ -1114,7 +1127,8 @@ def init(self, expr, min=1):
         self.min = min
         if len(expr.leaves) == 2:
             leaf_1 = expr.leaves[1]
-            if (leaf_1.has_form('List', 1, 2) and all(leaf.get_int_value() for leaf in leaf_1.leaves)):
+            allnumbers = all(not (leaf.get_int_value() is None) for leaf in leaf_1.get_leaves())
```

Comment:
`leaf.get_int_value() is not None` is more idiomatic than `not (leaf.get_int_value() is None)`